### PR TITLE
Add branch/release dev info in README

### PR DESCRIPTION
**Pull request by bvdvecht**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,13 +107,34 @@ explore a commercial market, please contact us for a license agreement.
...
 
-## Development
+# Development
 For code formatting, [`black`](https://github.com/psf/black) and [`isort`](https://github.com/PyCQA/isort) are used.
 Type hints should be added as much as possible.
 Types are checked using [`mypy`](https://github.com/python/mypy).
 
 Before code is pushed, make sure that the `make lint` command succeeds, which runs `black`, `isort`, `flake8` and `mypy`.
 
+## Branches and releases
```
Comment (platelminto) [inline README.md]: Maybe mention that these steps are just for maintainers? Since the repo is public.

File: README.md [hunk 2]
```diff
@@ -107,13 +107,34 @@ explore a commercial market, please contact us for a license agreement.
...
+A form of "git flow" is used for branch and release management. The main active branch is `develop`.
+New features are developed in new separate branches, preferrably with a name representing the new feature.
+To get the new features in the main branch, open a pull request for merging the feature branch into the `develop` branch.
+A `master` or `main` branch is not used.
+
+When a release is made, a new branch `release-X.Y` (e.g. `release-0.12`) is created from the `develop` branch.
+Only small fixes (patches) may be pushed onto this release branch. Bigger new features need to go into separate branches, merged into `develop`, and will hence end up in a later release.
+**Tags** are *only* applied on commits in the release branch. The first tag on a new release branch needs to be `vX.Y.0`, e.g. `v0.12.0`.
+Patches (i.e. commits on the release branch) may then be tagged with `vX.Y.1`, `vX.Y.2` etc.
```
Comment (platelminto) [inline README.md:129]: When are patches expected to be merged into `develop`?

File: README.md (hunk 1 above)
Comment (bvdvecht) [reply README.md]: Good point, I updated the text accordingly. Also, I realized it's probably a good idea to restrict pushing tags (triggering PyPI publishing) to maintainers only. Any idea how we could do this?

File: README.md (hunk 2 above)
Comment (bvdvecht) [reply README.md:129]: Added info on this

File: README.md (hunk 1 above)
Comment (bvdvecht) [reply README.md]: Actually I think only maintainers are able to push tags anyway, so it's fine as it is. Am I correct?

Comment (platelminto) [reply README.md]: Yes that's correct, we can leave it as is.

**Review by platelminto — APPROVED** [review]

**Review by platelminto — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (platelminto) [reply README.md:129]: Got it, looks good.

**Pull request merged**